Add color utility class

Nightfirecat opened the pull request:
This adds `ColorUtil::colorTag`, `ColorUtil::prependColorTag`, and `ColorUtil::wrapWithColorTag`, and implements them in existing plugins, as well as moving some color-related functions from `SwingUtil` into this new class.

Uses of `<col` in strings have been maintained as it is entirely proprietary to our code. (specifically the chat notifications plugin) Some uses of `<col=` ([1](https://github.com/Nightfirecat/runelite/blob/37407976a529cffc4cf34ee8fef5216c750a76b2/runelite-client/src/main/java/net/runelite/client/plugins/itemcharges/ItemChargePlugin.java#L53), [2](https://github.com/Nightfirecat/runelite/blob/37407976a529cffc4cf34ee8fef5216c750a76b2/runelite-client/src/main/java/net/runelite/client/plugins/kourendlibrary/KourendLibraryPlugin.java#L199), [3](https://github.com/Nightfirecat/runelite/blob/cfd06f7c9a855592982de318dbb4fed0f54edc6c/runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java#L93-L116), [4](https://github.com/Nightfirecat/runelite/blob/37407976a529cffc4cf34ee8fef5216c750a76b2/runelite-client/src/main/java/net/runelite/client/ui/overlay/components/TextComponent.java#L39)) in strings have been maintained for readability and have been moved as needed to constants.

Kamielvf commented on runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java:
```diff
@@ -525,9 +526,13 @@ private void addCounter()
 
 		BufferedImage taskImg = itemManager.getImage(itemSpriteId);
+		String taskTooltip = Text.prependColorTag("%s</br>", new Color(255, 119, 0))
+			+ Text.wrapWithColorTag("Pts:", new Color(255, 255, 0))
```
This is just `Color.YELLOW`

Kamielvf commented on runelite-client/src/main/java/net/runelite/client/util/Text.java:
```diff
@@ -62,4 +70,116 @@ else if (!inTag)
+	 * @param color The Color to create a tag from.
+	 * @return      A string of the color tag for the given `color`.
+	 */
+	public static String colorTag(Color color)
```
Since all color formatting should go trough this class, this can probably be private?

Nightfirecat replied:
It can't be made private as it is used [here](https://github.com/runelite/runelite/pull/4072/files#diff-bf265f5ad83eea9b469bdd43046c6ce2R137) and [here](https://github.com/runelite/runelite/pull/4072/files#diff-726aed9d5db03fc1de5694de6970f8b2R90).

Nightfirecat commented:
@deathbeam After some thought, I removed `Text::formatComparedText`. I agree now it was too specific.

deathbeam commented on runelite-client/src/main/java/net/runelite/client/util/Text.java:
```diff
@@ -24,11 +24,19 @@
  */
 public class Text
 {
+	private static final String DEFAULT_SEPARATOR = "/";
```
This DEFAULT_SEPARATOR and DEFAULT_COLOR looks unused now

Nightfirecat commented:
After some thought, I'm backing out 211870c as I'm working on something else which would better solve that issue.

Nightfirecat commented:
@deathbeam I've split the code into commits per plugin, a la #4346

Nightfirecat commented:
Added tests for the new functions since that was low-hanging fruit and fixed some bugs as a result. :+1:

Also rebased and applied the new functions to the recently-added achievement diary plugin.

Nightfirecat commented:
@deathbeam I reworked this PR to make a `ColorUtil` class instead, as that fits the bill a little better. Code is otherwise basically the same.
